test_analysis.py

```
@@ -0,0 +1,65 @@
+lib_files = get_dart_files('lib', ['.g.dart', '.freezed.dart'])
+test_files = get_dart_files('test')
+
+# Map lib files to test files based on name
+# Usually lib/path/to/file.dart -> test/path/to/file_test.dart
+untested_files = []
+tested_files = []
+
+for lib_file in lib_files:
+    # Handle abstract interfaces by name or content roughly
+    filename = os.path.basename(lib_file)
+    if filename.endswith('_repository.dart') or filename.endswith('_usecase.dart'):
```

Ruff (0.15.9)

[warning] Call `endswith` once with a `tuple`

Merge into a single `endswith` call

(PIE810)